batcher/batcher.mligo

```diff
@@ -1590,6 +1591,18 @@ let confirm_swap_pair_is_disabled_prior_to_removal
   (valid_swap:valid_swap) : unit =
   if valid_swap.is_disabled_for_deposits then () else failwith cannot_remove_swap_pair_that_is_not_disabled
 
+
+[@inline]
+let enforce_correct_side
+  (order:external_swap_order)
+  (valid_swap:valid_swap) : unit = 
+  let swap = order.swap in
+  if order.side = 0n then
+    if swap.from.token.name = valid_swap.swap.from.token.name then () else failwith incorrect_side_specified
+  else 
+    if swap.from.token.name = valid_swap.swap.to.name then () else failwith incorrect_side_specified
```

Comment:
Would be more obvious to enforce this after parsing the side to Buy or Sell. But ok.